primitives/src/light_client.rs

```diff
@@ -0,0 +1,85 @@
+use codec::{Decode, Encode};
+use frame_system::pallet_prelude::OriginFor;
+use scale_info::prelude::string::String;
+// use frame_support::dispatch::DispatchResult;
+use crate::{
+    gateway::GatewayABIConfig, ChainId, GatewayGenesisConfig, GatewayType, GatewayVendor,
+    TokenSysProps,
+};
+use scale_info::TypeInfo;
+use sp_runtime::DispatchError;
+use sp_std::vec::Vec;
+use t3rn_abi::{recode::Codec, types::Bytes};
+use t3rn_types::sfx::Sfx4bId;
+
+#[derive(Clone, Eq, Decode, Encode, PartialEq, Debug, TypeInfo)]
+pub struct LightClientHeartbeat<T: frame_system::Config> {
+    pub last_heartbeat: T::BlockNumber,
+    pub last_finalized_height: T::BlockNumber,
+    pub last_updated_height: T::BlockNumber,
+    pub is_halted: bool,
+}
+
+pub trait LightClient<T: frame_system::Config> {
+    fn get_latest_finalized_header(&self) -> Result<Option<Bytes>, DispatchError>;
+
+    fn get_latest_finalized_height(&self) -> Result<Option<T::BlockNumber>, DispatchError>;
+
+    fn get_latest_updated_height(&self) -> Result<Option<T::BlockNumber>, DispatchError>;
```

Comment:
Result<Option> is an anti pattern, i'd suggest doing Result<Enumofyourchoice> for varying states